project/ServerlessDeployPlugin.scala

```diff
@@ -17,20 +29,49 @@ object ServerlessDeployPlugin extends AutoPlugin {
 
   override def requires: Plugins = UniversalPlugin
 
+  def checkExitCode(exitCode: Int): Int = {
+    if (exitCode == 0) exitCode
+    else throw new IllegalStateException("Serverless returned a non-zero exit code. Please check the logs for more information.")
+  }
+
   override lazy val projectSettings = Seq(
-    serverlessDeployCommand := "serverless deploy --verbose".split(' ').toSeq,
+    serverlessPackageCommand := "serverless package --verbose".split(' ').toSeq,
+
     deploy := Def.inputTask {
-      import scala.sys.process._
+      val stage = Stage.parser.parsed.name.toLowerCase
+      val artifactPath = (Universal / packageBin).value.toString
 
-      val baseCommand = serverlessDeployCommand.value
-      val exitCode = Process(
-        baseCommand ++ Seq("--stage", Stage.parser.parsed.name.toLowerCase),
+      val serverlessExitCode = Process(
+        serverlessPackageCommand.value ++ Seq("--stage", stage),
         Option((ThisBuild / baseDirectory).value),
-        "ARTIFACT_PATH" -> (Universal / packageBin).value.toString,
+        "ARTIFACT_PATH" -> artifactPath,
       ).!
 
-      if (exitCode == 0) exitCode
-      else throw new IllegalStateException("Serverless returned a non-zero exit code. Please check the logs for more information.")
+      if (serverlessExitCode != 0) throw new IllegalStateException("Serverless returned a non-zero exit code. Please " +
+        "check the logs for more information.")
+
+      val stackName = Process(
+        s"serverless print --stage ${stage} --path provider.stackName --format text".split(' ').toSeq,
+        Option((ThisBuild / baseDirectory).value),
+        "ARTIFACT_PATH" -> artifactPath,
+      ).!!.trim
+
+      val cloudformationTemplate = (ThisBuild / baseDirectory).value /
+        ".serverless" / "cloudformation-template-update-stack.json"
+      IO.write(
+        cloudformationTemplate,
+        IO.read(cloudformationTemplate).replaceAll("CloudflareLambdaFunction", "Function")
+      )
+
+      val cloudformationExitCode = Process(
+        s"aws cloudformation deploy --template-file ${cloudformationTemplate.getPath}".split(' ').toSeq ++
```

Comment:
You may want to separate out the path from the parts that are `.split(' ')`, in case the path has any spaces in it. e.g.

```scala
(s"aws cloudformation deploy --template-file ".split(' ').toSeq :+ cloudformationTemplate.getPath) ++
s"--stack-name $stackName --capabilities CAPABILITY_NAMED_IAM".split(' ').toSeq
```

Reply:
Very good point. Fixed.